src/rez/utils/graph_utils.py

```diff
@@ -306,6 +306,133 @@ def _request_from_label(label):
     return label.strip('"').strip("'").rsplit('[', 1)[0]
 
 
+def _get_node_label(graph, node):
+    label_ = next(at[1] for at in graph.node_attr[node] if at[0] == "label")
+    return _request_from_label(label_)
+
+
+def _is_request_node(graph, node):
+    style = next(at[1] for at in graph.node_attr[node] if at[0] == "style")
+    return "dashed" in style
+
+
+def _iter_init_request_nodes(graph):
+    request_color = "#FFFFAA"  # NOTE: the color code is hardcoded in `solver`
+    for node, attrs in graph.node_attr.items():
+        for at in attrs:
+            if at[0] == "fillcolor" and at[1] == request_color:
+                yield node
+
+
+def failure_detail_from_graph(graph):
```

Comment:
I think we need to move the functions here that are specific to rez's solve graph, into `utils/resolve_graph.py`. The existing functions in this file were all generic graph util functions (not specific to graphs that rez produces), so it would be better to keep them separate.